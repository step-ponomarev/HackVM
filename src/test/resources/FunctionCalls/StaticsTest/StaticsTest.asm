@256
D=A
@SP
M=D
@Sys.init$ret.0
D=A
@SP
A=M
M=D
@SP
M=M+1
@LCL
D=M
@SP
A=M
M=D
@SP
M=M+1
@ARG
D=M
@SP
A=M
M=D
@SP
M=M+1
@THIS
D=M
@SP
A=M
M=D
@SP
M=M+1
@THAT
D=M
@SP
A=M
M=D
@SP
M=M+1

@SP
D=M

@5
D=D-A

@0
D=D-A

@ARG
M=D

@SP
D=M

@LCL
M=D

@Sys.init
0;JMP
(Sys.init$ret.0)
(Class1.set)
@ARG
D=M
@0
A=D+A
D=M
@SP
A=M
M=D
@SP
M=M+1
@Class1.0
D=A
@0
D=D+A
@ADDR_SAVE
M=D
@SP
M=M-1
A=M
D=M
@ADDR_SAVE
A=M
M=D
@ARG
D=M
@1
A=D+A
D=M
@SP
A=M
M=D
@SP
M=M+1
@Class1.1
D=A
@0
D=D+A
@ADDR_SAVE
M=D
@SP
M=M-1
A=M
D=M
@ADDR_SAVE
A=M
M=D
@0
D=A
@SP
A=M
M=D
@SP
M=M+1
@LCL
D=M
@frame
M=D
@5
D=A
@frame
A=M-D
D=M
@returnAddr
M=D
@SP
M=M-1
A=M
D=M
@ARG
A=M
M=D
@ARG
D=M+1
@SP
M=D
@1
D=A
@frame
A=M-D
D=M
@THAT
M=D
@2
D=A
@frame
A=M-D
D=M
@THIS
M=D
@3
D=A
@frame
A=M-D
D=M
@ARG
M=D
@4
D=A
@frame
A=M-D
D=M
@LCL
M=D
@returnAddr
A=M
0;JMP
(Class1.get)
@Class1.0
D=A
@0
A=D+A
D=M
@SP
A=M
M=D
@SP
M=M+1
@Class1.1
D=A
@0
A=D+A
D=M
@SP
A=M
M=D
@SP
M=M+1
@SP
M=M-1
A=M
D=M
@ARG1
M=D
@SP
M=M-1
A=M
D=M
@ARG2
M=D
@ARG2
D=M
@ARG1
A=M
D=D-A
@SP
A=M
M=D
@SP
M=M+1
@LCL
D=M
@frame
M=D
@5
D=A
@frame
A=M-D
D=M
@returnAddr
M=D
@SP
M=M-1
A=M
D=M
@ARG
A=M
M=D
@ARG
D=M+1
@SP
M=D
@1
D=A
@frame
A=M-D
D=M
@THAT
M=D
@2
D=A
@frame
A=M-D
D=M
@THIS
M=D
@3
D=A
@frame
A=M-D
D=M
@ARG
M=D
@4
D=A
@frame
A=M-D
D=M
@LCL
M=D
@returnAddr
A=M
0;JMP
(Sys.init)
@6
D=A
@SP
A=M
M=D
@SP
M=M+1
@8
D=A
@SP
A=M
M=D
@SP
M=M+1
@Class1.set$ret.0
D=A
@SP
A=M
M=D
@SP
M=M+1
@LCL
D=M
@SP
A=M
M=D
@SP
M=M+1
@ARG
D=M
@SP
A=M
M=D
@SP
M=M+1
@THIS
D=M
@SP
A=M
M=D
@SP
M=M+1
@THAT
D=M
@SP
A=M
M=D
@SP
M=M+1

@SP
D=M

@5
D=D-A

@2
D=D-A

@ARG
M=D

@SP
D=M

@LCL
M=D

@Class1.set
0;JMP
(Class1.set$ret.0)
@5
D=A
@0
D=D+A
@ADDR_SAVE
M=D
@SP
M=M-1
A=M
D=M
@ADDR_SAVE
A=M
M=D
@23
D=A
@SP
A=M
M=D
@SP
M=M+1
@15
D=A
@SP
A=M
M=D
@SP
M=M+1
@Class2.set$ret.0
D=A
@SP
A=M
M=D
@SP
M=M+1
@LCL
D=M
@SP
A=M
M=D
@SP
M=M+1
@ARG
D=M
@SP
A=M
M=D
@SP
M=M+1
@THIS
D=M
@SP
A=M
M=D
@SP
M=M+1
@THAT
D=M
@SP
A=M
M=D
@SP
M=M+1

@SP
D=M

@5
D=D-A

@2
D=D-A

@ARG
M=D

@SP
D=M

@LCL
M=D

@Class2.set
0;JMP
(Class2.set$ret.0)
@5
D=A
@0
D=D+A
@ADDR_SAVE
M=D
@SP
M=M-1
A=M
D=M
@ADDR_SAVE
A=M
M=D
@Class1.get$ret.0
D=A
@SP
A=M
M=D
@SP
M=M+1
@LCL
D=M
@SP
A=M
M=D
@SP
M=M+1
@ARG
D=M
@SP
A=M
M=D
@SP
M=M+1
@THIS
D=M
@SP
A=M
M=D
@SP
M=M+1
@THAT
D=M
@SP
A=M
M=D
@SP
M=M+1

@SP
D=M

@5
D=D-A

@0
D=D-A

@ARG
M=D

@SP
D=M

@LCL
M=D

@Class1.get
0;JMP
(Class1.get$ret.0)
@Class2.get$ret.0
D=A
@SP
A=M
M=D
@SP
M=M+1
@LCL
D=M
@SP
A=M
M=D
@SP
M=M+1
@ARG
D=M
@SP
A=M
M=D
@SP
M=M+1
@THIS
D=M
@SP
A=M
M=D
@SP
M=M+1
@THAT
D=M
@SP
A=M
M=D
@SP
M=M+1

@SP
D=M

@5
D=D-A

@0
D=D-A

@ARG
M=D

@SP
D=M

@LCL
M=D

@Class2.get
0;JMP
(Class2.get$ret.0)
(Sys.init$END)
@Sys.init$END
0;JMP
(Class2.set)
@ARG
D=M
@0
A=D+A
D=M
@SP
A=M
M=D
@SP
M=M+1
@Class2.0
D=A
@0
D=D+A
@ADDR_SAVE
M=D
@SP
M=M-1
A=M
D=M
@ADDR_SAVE
A=M
M=D
@ARG
D=M
@1
A=D+A
D=M
@SP
A=M
M=D
@SP
M=M+1
@Class2.1
D=A
@0
D=D+A
@ADDR_SAVE
M=D
@SP
M=M-1
A=M
D=M
@ADDR_SAVE
A=M
M=D
@0
D=A
@SP
A=M
M=D
@SP
M=M+1
@LCL
D=M
@frame
M=D
@5
D=A
@frame
A=M-D
D=M
@returnAddr
M=D
@SP
M=M-1
A=M
D=M
@ARG
A=M
M=D
@ARG
D=M+1
@SP
M=D
@1
D=A
@frame
A=M-D
D=M
@THAT
M=D
@2
D=A
@frame
A=M-D
D=M
@THIS
M=D
@3
D=A
@frame
A=M-D
D=M
@ARG
M=D
@4
D=A
@frame
A=M-D
D=M
@LCL
M=D
@returnAddr
A=M
0;JMP
(Class2.get)
@Class2.0
D=A
@0
A=D+A
D=M
@SP
A=M
M=D
@SP
M=M+1
@Class2.1
D=A
@0
A=D+A
D=M
@SP
A=M
M=D
@SP
M=M+1
@SP
M=M-1
A=M
D=M
@ARG1
M=D
@SP
M=M-1
A=M
D=M
@ARG2
M=D
@ARG2
D=M
@ARG1
A=M
D=D-A
@SP
A=M
M=D
@SP
M=M+1
@LCL
D=M
@frame
M=D
@5
D=A
@frame
A=M-D
D=M
@returnAddr
M=D
@SP
M=M-1
A=M
D=M
@ARG
A=M
M=D
@ARG
D=M+1
@SP
M=D
@1
D=A
@frame
A=M-D
D=M
@THAT
M=D
@2
D=A
@frame
A=M-D
D=M
@THIS
M=D
@3
D=A
@frame
A=M-D
D=M
@ARG
M=D
@4
D=A
@frame
A=M-D
D=M
@LCL
M=D
@returnAddr
A=M
0;JMP
