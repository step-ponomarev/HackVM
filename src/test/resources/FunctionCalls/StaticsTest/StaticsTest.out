| RAM[0] |RAM[261]|RAM[262]|
|     48 |      6 |      8 |
